googleapiclient/http.py

```diff
@@ -91,12 +95,24 @@ def _retry_request(http, num_retries, req_type, sleep, rand, uri, method, *args,
 
     try:
       resp, content = http.request(uri, method, *args, **kwargs)
-    except ssl.SSLError:
+    except (ssl.SSLError, TimeoutError) as ex:
       if retry_num == num_retries:
         raise
       else:
+        logging.warning('Encountered exception %s', repr(ex))
         continue
-    if resp.status < 500:
+    reason = ''
+    try:
+      if resp and resp.status == 403:
+        data = json.loads(content.decode('utf-8'))
+        reason = data['error']['errors'][0]['reason']
+        logging.warning('Encountered status %d with reason "%s"' %
+                        (resp.status, reason))
+    except Exception:
+      pass
+    if resp.status < 500 and not ((resp.status == 403 and
```

Comment:
This is a bit difficult to follow. Could you break it into multiple if statements?